### PR TITLE
Allow scripts to get modes

### PR DESCRIPTION
**Pull request by tatsuy**
This PR allow LUA scripting change the flight mode.
I referenced #12575

I checked this in SITL (Plane, Copter, Sub, Rover, AntennaTracker) and Cube (with Copter) using the following script

```lua
function update()
  mode = vehicle:get_mode()
  gcs:send_text(0, mode)
  return update, 2000
end

return update()
```

### CODE REVIEW

**Comment by tatsuy** [comment]
image

**Comment by rmackay9** [comment]
@WickedShell, I've tested this on a CubeBlack and it worked correctly.  I also ran "make run" from the AP_Scripting/generator directory to ensure that no accidental manual changes had been made to the lua_generated_bindings files.

File: ArduCopter/Copter.h [hunk 1]
```diff
@@ -796,6 +796,7 @@ class Copter : public AP_Vehicle {
     // mode.cpp
     bool set_mode(Mode::Number mode, ModeReason reason);
     bool set_mode(const uint8_t new_mode, const ModeReason reason) override;
+    uint8_t get_mode() const override { return (uint8_t)control_mode; }
```
Comment (rmackay9) [inline ArduCopter/Copter.h]: we discussed on the call that we should probably make this use the control->mode_number() like the other vehicles (if possible).

Comment (tatsuy) [reply ArduCopter/Copter.h]: I'm done. #13431

**Pull request merged**